src/Microsoft.AspNet.OData/Adapters/WebApiRequestMessage.cs

```diff
@@ -116,14 +116,27 @@ public ODataDeserializerProvider DeserializerProvider
             get { return this.innerRequest.GetDeserializerProvider(); }
         }
 
+        /// <summary>
+        /// Get the next page link for a given page size.
+        /// </summary>
+        /// <param name="pageSize">The page size.</param>
+        /// <param name="skipTokenValue">value for the skiptoken</param>
+        /// <returns></returns>
+        public Uri GetNextPageLink(int pageSize, string skipTokenValue)
+        {
+            this.Context.NextLink = this.innerRequest.GetNextPageLink(pageSize, skipTokenValue);
+            return this.Context.NextLink;
+        }
+
         /// <summary>
         /// Get the next page link for a given page size.
         /// </summary>
         /// <param name="pageSize">The page size.</param>
         /// <returns></returns>
         public Uri GetNextPageLink(int pageSize)
         {
-            return this.innerRequest.GetNextPageLink(pageSize);
+            this.Context.NextLink = this.innerRequest.GetNextPageLink(pageSize);
```

Comment:
Help me understand this code; it looks like calling GetNextPageLink has the side-effect of setting the NextLink on the context, but if I don't call GetNextPageLink then the NextLink doesn't get set on the context?

Reply:
Prior to my change, Context.NextLink did not get set if GetNextPageLink method got called. The consumer of the API had to explicitly set that after making a call to this public method. To make this paging API easier to consume, I decided to move the setting of the Context.NextLink inside the public method as I thought it is not intuitive for consumers to set the context.nextlink explicitly.